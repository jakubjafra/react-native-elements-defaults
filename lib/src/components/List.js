const utils = require('../utils');
const functionalComponentCompose = require('../compose/funcionalComponent');

const COMPOSE_NAME = 'List';
const compose = functionalComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/list/List'),
        compose,
        true,
    ),
};
