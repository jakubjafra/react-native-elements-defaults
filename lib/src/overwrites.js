const Avatar = require('./components/Avatar');
const Badge = require('./components/Badge');
const Button = require('./components/Button');
const ButtonGroup = require('./components/ButtonGroup');
const Card = require('./components/Card');
const CheckBox = require('./components/CheckBox');
const colors = require('./files/colors');
const Divider = require('./components/Divider');
const FeaturedTile = require('./components/FeaturedTile');
const fonts = require('./files/fonts');
const FormInput = require('./components/FormInput');
const FormLabel = require('./components/FormLabel');
const FormValidationMessage = require('./components/FormValidationMessage');
const Header = require('./components/Header');
const Icon = require('./components/Icon');
const List = require('./components/List');
const ListItem = require('./components/ListItem');
const PricingCard = require('./components/PricingCard');
const Rating = require('./components/Rating');
const Search = require('./components/Search');
const Slider = require('./components/Slider');
const SocialIcon = require('./components/SocialIcon');
const Text = require('./components/Text');
const Tile = require('./components/Tile');

module.exports = {
    [Avatar.name]: Avatar,
    [Badge.name]: Badge,
    [Button.name]: Button,
    [ButtonGroup.name]: ButtonGroup,
    [Card.name]: Card,
    [CheckBox.name]: CheckBox,
    [colors.name]: colors,
    [Divider.name]: Divider,
    [FeaturedTile.name]: FeaturedTile,
    [fonts.name]: fonts,
    [FormInput.name]: FormInput,
    [FormLabel.name]: FormLabel,
    [FormValidationMessage.name]: FormValidationMessage,
    [Header.name]: Header,
    [Icon.name]: Icon,
    [List.name]: List,
    [ListItem.name]: ListItem,
    [PricingCard.name]: PricingCard,
    [Rating.name]: Rating,
    [Search.name]: Search,
    [Slider.name]: Slider,
    [SocialIcon.name]: SocialIcon,
    [Text.name]: Text,
    [Tile.name]: Tile,
};
