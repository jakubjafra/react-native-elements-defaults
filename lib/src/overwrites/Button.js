const utils = require('../utils');
const functionalComponentCompose = require('./_funcionalComponentCompose');

const COMPOSE_NAME = 'Button';
const compose = functionalComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/buttons/Button'),
        compose,
        true,
    ),
};
