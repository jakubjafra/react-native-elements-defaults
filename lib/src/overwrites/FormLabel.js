const utils = require('../utils');
const classComponentCompose = require('./_classComponentCompose');

const COMPOSE_NAME = 'FormLabel';
const compose = classComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/form/FormLabel'),
        compose,
        true,
    ),
};
