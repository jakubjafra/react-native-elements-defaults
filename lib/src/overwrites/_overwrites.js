const colors = require('./colors');
const fonts = require('./fonts');
const Text = require('./Text');

module.exports = {
    [colors.name]: colors,
    [fonts.name]: fonts,
    [Text.name]: Text,
};
