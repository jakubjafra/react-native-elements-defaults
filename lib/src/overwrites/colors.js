const utils = require('../utils');
const fileCompose = require('./_fileCompose');

const COMPOSE_NAME = '$colors';
const compose = fileCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/config/colors'),
        compose,
        false,
    ),
};
