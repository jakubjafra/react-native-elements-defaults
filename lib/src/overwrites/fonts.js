const utils = require('../utils');
const fileCompose = require('./_fileCompose');

const COMPOSE_NAME = '$fonts';
const compose = fileCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/config/fonts'),
        compose,
        false,
    ),
};
