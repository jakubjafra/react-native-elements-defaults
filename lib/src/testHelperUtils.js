import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name,
                      __oldElement__,
                      __element__,
                      basicTheme,
                      createElement = element => React.createElement(element, null, null)) {
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
