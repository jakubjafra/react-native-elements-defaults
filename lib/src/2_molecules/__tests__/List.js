import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import * as utils from '../../testHelperUtils';

describe('List', () => {
    utils.defaultTests(
        'List',
        require('react-native-elements/src/list/List'),
        require('../../overwrites/List'),
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );
});
