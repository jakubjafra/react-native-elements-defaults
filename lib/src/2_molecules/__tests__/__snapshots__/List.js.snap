// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List with basic theme applied List renders correctly 1`] = `
<View
  avatarContainerStyle={undefined}
  avatarOverlayContainerStyle={undefined}
  avatarStyle={undefined}
  disabledStyle={undefined}
  rightTitleContainerStyle={undefined}
  rightTitleStyle={undefined}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  subtitleContainerStyle={undefined}
  subtitleStyle={undefined}
  textInputContainerStyle={undefined}
  textInputStyle={undefined}
  titleContainerStyle={undefined}
  titleStyle={undefined}
  wrapperStyle={undefined}
/>
`;

exports[`List with basic theme applied OldList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;

exports[`List with no theme applied List renders correctly 1`] = `
<View
  avatarContainerStyle={undefined}
  avatarOverlayContainerStyle={undefined}
  avatarStyle={undefined}
  disabledStyle={undefined}
  rightTitleContainerStyle={undefined}
  rightTitleStyle={undefined}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
  subtitleContainerStyle={undefined}
  subtitleStyle={undefined}
  textInputContainerStyle={undefined}
  textInputStyle={undefined}
  titleContainerStyle={undefined}
  titleStyle={undefined}
  wrapperStyle={undefined}
/>
`;

exports[`List with no theme applied OldList and List returns the same thing if no theme is applied 1`] = `
"[32m- Expected[39m
[31m+ Received[39m

[2m  <View[22m
[32m-   avatarContainerStyle={undefined}[39m
[32m-   avatarOverlayContainerStyle={undefined}[39m
[32m-   avatarStyle={undefined}[39m
[32m-   disabledStyle={undefined}[39m
[32m-   rightTitleContainerStyle={undefined}[39m
[32m-   rightTitleStyle={undefined}[39m
[2m    style={[22m
[2m      Array [[22m
[2m        Object {[22m
[2m          \\"backgroundColor\\": \\"#ffffff\\",[22m
[2m          \\"borderColor\\": \\"#bbb\\",[22m
[2m          \\"borderTopWidth\\": 1,[22m
[2m          \\"marginTop\\": 20,[22m
[2m        },[22m
[2m        undefined,[22m
[2m      ][22m
[2m    }[22m
[32m-   subtitleContainerStyle={undefined}[39m
[32m-   subtitleStyle={undefined}[39m
[32m-   textInputContainerStyle={undefined}[39m
[32m-   textInputStyle={undefined}[39m
[32m-   titleContainerStyle={undefined}[39m
[32m-   titleStyle={undefined}[39m
[32m-   wrapperStyle={undefined}[39m
[2m  />[22m"
`;

exports[`List with no theme applied OldList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;
